test/vm_test.exs

```diff
@@ -1456,6 +1519,189 @@ defmodule Signet.VmTest do
+        :mload,
+        :stop
+      ],
+      exp_error: {:unknown_ffi, <<1::160>>}
+    },
+    %{
+      name: "Static Call - Add",
+      ffis: %{
+        <<2::160>> => &FFI.add_ffi/1
+      },
+      code: [
+        {:push, 32, word("0x771602f700000000000000000000000000000000000000000000000000000000")},
+        {:push, 32, word(0x100)},
+        :mstore,
+        {:push, 32, word("0x0000000000000000000000000000000000000000000000000000000000000010")},
+        {:push, 32, word(0x104)},
+        :mstore,
+        {:push, 32, word("0x000000000000000000000000000000000000000000000000000000000000002a")},
+        {:push, 32, word(0x124)},
+        :mstore,
+        # ret_size
+        {:push, 32, word(32)},
+        # ret_offset
+        {:push, 32, word(0)},
+        # args_size
+        {:push, 32, word(0x44)},
+        # args_offset
+        {:push, 32, word(0x100)},
+        # ffi address
+        {:push, 32, <<2::256>>},
+        :gas,
+        :staticcall,
+        {:push, 32, word(0)},
+        :mload,
+        :stop
+      ],
+      exp_stack: [
+        word(0x3A)
+      ]
+    },
+    %{
+      name: "Return Data Size",
+      ffis: %{
+        <<3::160>> => &FFI.long_ffi/1
+      },
+      code: [
+        {:push, 32, word("0x1122334400000000000000000000000000000000000000000000000000000000")},
+        {:push, 32, word(100)},
+        :mstore,
+        # ret_size
+        {:push, 32, word(32)},
+        # ret_offset
+        {:push, 32, word(0)},
+        # args_size
+        {:push, 32, word(4)},
+        # args_offset
+        {:push, 32, word(100)},
+        # ffi address
+        {:push, 32, <<3::256>>},
+        :gas,
+        :staticcall,
+        :returndatasize,
+        :stop
+      ],
+      exp_stack: [
+        word(90)
+      ]
+    },
+    %{
+      name: "Return Data Copy",
+      ffis: %{
+        <<3::160>> => &FFI.long_ffi/1
+      },
+      code: [
+        {:push, 32, word("0x1122334400000000000000000000000000000000000000000000000000000000")},
+        {:push, 32, word(100)},
+        :mstore,
+        # ret_size
+        {:push, 32, word(32)},
+        # ret_offset
+        {:push, 32, word(0)},
+        # args_size
+        {:push, 32, word(4)},
+        # args_offset
+        {:push, 32, word(100)},
+        # ffi address
+        {:push, 32, <<3::256>>},
+        :gas,
+        :staticcall,
+        # size
+        {:push, 1, word(90, 1)},
+        # offset
+        {:push, 1, word(1, 1)},
+        # dest_offset
+        {:push, 1, word(200, 1)},
+        :returndatacopy,
+        {:push, 32, word(200)},
+        :mload,
+        {:push, 32, word(232)},
+        :mload,
+        {:push, 32, word(264)},
+        :mload,
+        :stop
+      ],
+      exp_stack: [
+        ~h[0x66778899aabbccddeeff112233445566778899aabbccddeeff00000000000000],
+        ~h[0x445566778899aabbccddeeff112233445566778899aabbccddeeff1122334455],
+        ~h[0x2233445566778899aabbccddeeff112233445566778899aabbccddeeff112233]
```

Comment:
nit: this is slightly confusing to verify because there's repeating sequences in the larger byte array